src/app/clusters/thread-network-directory-server/DefaultThreadNetworkDirectoryStorage.cpp

```diff
@@ -118,11 +118,18 @@ CHIP_ERROR DefaultThreadNetworkDirectoryStorage::RemoveNetwork(const ExtendedPan
     VerifyOrReturnError(FindNetwork(exPanId, index), CHIP_ERROR_NOT_FOUND);
 
     // Move subsequent elements down to fill the deleted slot
-    for (mCount--; index < mCount; index++)
+    index_t subsequent = --mCount - index;
+    auto * element     = &mExtendedPanIds[index];
+    memmove(element, element + 1, subsequent * sizeof(*element));
```

Comment:
Do we have a static_assert somewhere that the type is safe to memmove?  Ideally that static_assert would be right here.

Reply:
Yes, this is documented and static_asserted near the definition of the ExtendedPanId type.